pkg/agent/ipassigner/ip_assigner_linux.go

```diff
@@ -83,9 +95,47 @@ func NewIPAssigner(nodeTransportInterface string, dummyDeviceName string) (*ipAs
 			return nil, fmt.Errorf("error when loading IP addresses from the system: %v", err)
 		}
 	}
+	// Copy the assigned IPs on the dummy interface to ARP responder or NDP responder.
+	var assignedIPsV4, assignedIPsV6 []net.IP
+	for assigned := range a.assignedIPs {
+		ip := net.ParseIP(assigned)
+		if utilnet.IsIPv4(ip) {
+			assignedIPsV4 = append(assignedIPsV4, ip)
+		}
+		if utilnet.IsIPv6(ip) {
+			assignedIPsV6 = append(assignedIPsV6, ip)
+		}
+	}
+	if a.arpResponder != nil {
+		if err = a.arpResponder.ReplaceIPs(assignedIPsV4); err != nil {
```

Comment:
I meant adding `ReplaceIPs` to `IPAssigner` and calling it in `removeStaleEgressIPs` so it will broadcast correct IPs once and not stale IPs.

Reply:
Got it. Sorry for the misunderstanding.
It seems a little obscure that the IPAssigner can only work as expected after ReplaceIPs is called. Otherwise, the responders will not have any assigned IPs by default. What do you think?

Reply:
then how about naming it "InitIPs"? perhaps we should also move `loadIPAddresses` to it so the method is responsible for cleaning stale IPs assigned to the dummy interface and broadcast correct IPs.

Reply:
done. thanks!